src/software/proto/message_translation/ssl_detection_message_translator.cpp

```diff
@@ -0,0 +1,87 @@
+#include "software/proto/message_translation/ssl_detection_message_translator.h"
+
+#include "shared/constants.h"
+
+std::unique_ptr<SSL_DetectionBall> createDetectionBall(const BallState& ball)
+{
+    auto detection_ball = std::make_unique<SSL_DetectionBall>();
+
+    // Give all detections "perfect" confidence since we have no means of reasonably
+    // esimating confidence
+    detection_ball->set_confidence(1.0);
+    // We don't have the information to simulate a camera and therefore the area of the
+    // ball
+    detection_ball->clear_area();
+    auto x_position_mm = static_cast<float>(ball.position().x() * MILLIMETERS_PER_METER);
+    auto y_position_mm = static_cast<float>(ball.position().y() * MILLIMETERS_PER_METER);
+    detection_ball->set_x(x_position_mm);
+    detection_ball->set_y(y_position_mm);
+    detection_ball->set_z(static_cast<float>(ball.height() * MILLIMETERS_PER_METER));
+    // grSim uses the ball's position as its pixel values, so we do the same thing here
+    // since we don't have the information to simulate a camera
+    detection_ball->set_pixel_x(x_position_mm);
+    detection_ball->set_pixel_y(y_position_mm);
+
+    return std::move(detection_ball);
+}
+
+std::unique_ptr<SSL_DetectionRobot> createDetectionRobot(const RobotStateWithId& robot)
+{
+    auto detection_robot = std::make_unique<SSL_DetectionRobot>();
+
+    // Give all detections "perfect" confidence since we have no means of reasonably
+    // esimating confidence
+    detection_robot->set_confidence(1.0);
+    detection_robot->set_robot_id(robot.id);
+    auto x_position_mm =
+        static_cast<float>(robot.robot_state.position().x() * MILLIMETERS_PER_METER);
+    auto y_position_mm =
+        static_cast<float>(robot.robot_state.position().y() * MILLIMETERS_PER_METER);
+    detection_robot->set_x(x_position_mm);
+    detection_robot->set_y(y_position_mm);
+    detection_robot->set_orientation(
+        static_cast<float>(robot.robot_state.orientation().toRadians()));
+    // grSim uses the robot's position as its pixel values, so we do the same thing here
+    // since we don't have the information to simulate a camera
+    detection_robot->set_pixel_x(x_position_mm);
+    detection_robot->set_pixel_y(y_position_mm);
+    detection_robot->set_height(
+        static_cast<float>(robot.robot_state.height() * MILLIMETERS_PER_METER));
```

Comment:
I'm a bit conflicted on adding "height" to robot state. On one hand it's nice to inject everything like this, but I really do not like default-value arguments, and right now we're basically just setting all robot heights to a constant value, and I don't ever see that changing. So I think I'd prefer we just use a constant for the height here if possible.

Reply:
* What if the height didn't have a default value? I just did that since it's the lowest friction way of adding it initially.
* I initially did it with the constant but didn't like testing it due to the "magic number" aspect of the constant.
* Do you have similar concerns about the `height` in `BallState`?

Reply:
ok I removed the `height` from robot state and use the constant in the translation.